engine/src/main/java/org/terasology/rendering/dag/nodes/AlphaRejectBlocksNode.java

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.dag.nodes;
+
+import org.lwjgl.opengl.GL11;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.engine.ComponentSystemManager;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.monitoring.PerformanceMonitor;
+import org.terasology.registry.In;
+import org.terasology.rendering.assets.material.Material;
+import org.terasology.rendering.assets.shader.ShaderProgramFeature;
+import org.terasology.rendering.cameras.Camera;
+import org.terasology.rendering.dag.WireframeCapableNode;
+import org.terasology.rendering.dag.stateChanges.EnableMaterial;
+import org.terasology.rendering.dag.stateChanges.SetViewportToSizeOf;
+import org.terasology.rendering.primitives.ChunkMesh;
+import org.terasology.rendering.world.RenderQueuesHelper;
+import org.terasology.rendering.world.WorldRenderer;
+import org.terasology.world.chunks.ChunkConstants;
+import org.terasology.world.chunks.RenderableChunk;
+
+import static org.terasology.rendering.opengl.DefaultDynamicFBOs.READ_ONLY_GBUFFER;
+
+/**
+ * This node uses alpha-rejection to render semi-transparent blocks (i.e. tree foliage) and
+ * semi-transparent billboards (i.e. plants on the ground).
+ *
+ * Alpha-rejection is the idea that if a fragment has an alpha value lower than some threshold
+ * it gets discarded, leaving the color already stored in the frame buffer untouched.
+ *
+ * This is a less expensive way to render semi-transparent objects compared to alpha-blending.
+ * In alpha-blending the color of a semi-transparent fragment is combined with
+ * the color stored in the frame buffer and the resulting color overwrites the previously stored one.
+ */
+public class AlphaRejectBlocksNode extends WireframeCapableNode {
```

Comment:
This class has been renamed from ChunksAlphaRejectNode. Technically the class does iterate through chunks and retrieves from them the mesh to be rendered. However, the mesh only contains the geometry for some of the blocks, the blocks that are semi-transparent. I also feel is more intuitive to think that this node renders blocks rather than chunks, as chunks per se do not have as much meaning from a visual perspective.